kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsService.java

```diff
@@ -35,8 +40,12 @@ public record Config(URI vaultUrl,
 
     @NonNull
     @Override
-    public VaultKms buildKms(Config options) {
-        return new VaultKms(options.vaultUrl(), options.vaultToken(), Duration.ofSeconds(20));
+    public VaultKms buildKms(Config options, FilterFactoryContext context) {
```

Comment:
Parnas: do we really want to give the whole FFC to the KMS?  How about an SSLContextProvider interface which is implemented by FFC? SSLContextProvider would need to be API surface of course.